mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -1212,11 +1217,15 @@ def compute_next_token(current_size):
             cases_orderby, parsed_orderby, sorting_joins = _get_orderby_clauses(order_by, session)
 
             stmt = select(SqlRun, *cases_orderby)
-            attribute_filters, non_attribute_filters = _get_sqlalchemy_filter_clauses(
-                parsed_filters, session, self._get_dialect()
-            )
+            (
+                attribute_filters,
+                non_attribute_filters,
+                dataset_filters,
+            ) = _get_sqlalchemy_filter_clauses(parsed_filters, session, self._get_dialect())
             for non_attr_filter in non_attribute_filters:
                 stmt = stmt.join(non_attr_filter)
+            for dataset_filter in dataset_filters:
+                stmt = stmt.join(dataset_filter, text("runs.run_uuid = anon_1.destination_id"))
```

Comment:
Is it possible to alias `anon_1` so that we can refer to it as `inputs` here? No worries if not - at minimum, can we add an inline comment explaining what's going on here and link to SQLAlchemy reference docs?